### PR TITLE
Updated json response to match spec

### PR DESCRIPTION
**Pull request by davebryson**
Should address:
go-rpc issue #16 
tendermint issue #485

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -42,24 +42,44 @@ type Result interface {
 
 //----------------------------------------
 
+type Error struct {
```
Comment (melekes) [inline types/types.go]: Can't we make it one error type? I don't see where `jsonRpcError.Data` field is used.

Comment (davebryson) [inline types/types.go]: Yes.  The Data field is not currently used, but it is part of the spec (but not required).

**Comment by davebryson** [comment]
Removed redundant Error. Added Error "helper" functions

**Comment by melekes** [comment]
https://github.com/tendermint/tendermint/pull/521

**Pull request closed without merging**